res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -0,0 +1,412 @@
+const Prime4ColorMapper = new ColorMapper({
+    //  0x000055: 0x01, // dark blue
+    //  0x0000AA: 0x02, // dim blue
+    0x0000FF: 0x03, // blue
+    //  0x005500: 0x04, // dark green
+    //  0x005555: 0x05, // dark cyan
+    //  0x0055AA: 0x06,
+    //  0x0055FF: 0x07,
+    //  0x00AA00: 0x08, // dim green
+    //  0x00AA55: 0x09,
+    //  0x00AAAA: 0x0A, // dim cyan
+    //  0x00AAFF: 0x0B,
+    0x00FF00: 0x0C, // green
+    //  0x00FF55: 0x0D,
+    //  0x00FFAA: 0x0E,
+    0x00FFFF: 0x0F, // cyan
+    //  0x550000: 0x10, // dark red
+    //  0x550055: 0x11, // dark magenta
+    //  0x5500AA: 0x12,
+    //  0x5500FF: 0x13,
+    //  0x5555AA: 0x16,
+    //  0x5555FF: 0x17,
+    //  0x55AA00: 0x18,
+    //  0x55AA55: 0x19,
+    //  0x55AAAA: 0x1A,
+    //  0x55AAFF: 0x1B,
+    //  0x55FF00: 0x1C,
+    //  0x55FF55: 0x1D,
+    //  0x55FFAA: 0x1E,
+    //  0x55FFFF: 0x1F,
+    //  0xAA0000: 0x20, // dim red
+    //  0xAA0055: 0x21,
+    //  0xAA00AA: 0x22, // dim magenta
+    //  0xAA00FF: 0x23,
+    //  0xAA5500: 0x24,
+    //  0xAA5555: 0x25,
+    //  0xAA55AA: 0x26,
+    //  0xAA55FF: 0x27,
+    //  0xAAAA00: 0x28, // dim yellow
+    //  0xAAAA55: 0x29,
+    //  0xAAAAAA: 0x2A, // dim white/grey
+    //  0xAAAAFF: 0x2B,
+    //  0xAAFF00: 0x2C,
+    //  0xAAFF55: 0x2D,
+    //  0xAAFFAA: 0x2E,
+    //  0xAAFFFF: 0x2F,
+    0xFF0000: 0x30, // red
+    //  0xFF0055: 0x31,
+    //  0xFF00AA: 0x32,
+    0xFF00FF: 0x33, // purple
+    0xFF5500: 0x34, // orange
+    //  0xFF5555: 0x35,
+    //  0xFF55AA: 0x36,
+    //  0xFF55FF: 0x37,
+    //  0xFFAA00: 0x38,
+    //  0xFFAA55: 0x39,
+    0xFFAAAA: 0x3A, // pink
+    //  0xFFAAFF: 0x3B,
+    0xFFFF00: 0x3C, // yellow
+    //  0xFFFF55: 0x3D,
+    //  0xFFFFAA: 0x3E,
+    0xFFFFFF: 0x3F, // white
+});
+
+const colDeck = [
+    Prime4.rgbCode[deckColors[0]],
+    Prime4.rgbCode[deckColors[1]],
+    Prime4.rgbCode[deckColors[2]],
+    Prime4.rgbCode[deckColors[3]],
+];
+const colDeckDark = [
+    Prime4.rgbCode[deckColors[0] + "Dark"],
+    Prime4.rgbCode[deckColors[1] + "Dark"],
+    Prime4.rgbCode[deckColors[2] + "Dark"],
+    Prime4.rgbCode[deckColors[3] + "Dark"],
+];
+
+// Register '0x9n' as a button press and 'ox8n' as a button release
+components.Button.prototype.isPress = function(channel, control, value, status) {
+    return (status & 0xF0) === 0x90;
+};
+
+Prime4.init = function(_id, _debug) {
+    // Turn off all LEDs
+    midi.sendShortMsg(0x90, 0x75, 0x00);
+
+    // Initialise all three physical sections of the unit
+    Prime4.leftDeck = new Prime4.Deck([1, 3], 4);
+    Prime4.rightDeck = new Prime4.Deck([2, 4], 5);
+    Prime4.mixerA = new mixerStrip(1, 1);
+    Prime4.mixerB = new mixerStrip(2, 2);
+    Prime4.mixerC = new mixerStrip(3, 3);
+    Prime4.mixerD = new mixerStrip(4, 4);
+
+    // Initial LED values to set (Hopefully these will automatically initialize, but for now they won't.)
+    midi.sendShortMsg(0x90, 0x0D, colDeckDark[0]);                    // PFL 1
```

Comment:
this is weird. You're doing everything correctly. Can you check if the `.trigger` method is being called upon initialization?